client/src/components/Profile/Sidebar.js

```diff
@@ -1,36 +1,125 @@
-import React from 'react';
-import { Grid, Typography } from '@material-ui/core';
-import { NavLink } from 'react-router-dom';
-import { makeStyles } from '@material-ui/styles';
-import profilePic from '../../static/images/profilePicSample.png';
+import React, { useState } from 'react';
+import { Grid, Typography, Badge, Divider, Avatar, Button } from '@material-ui/core';
+import { NavLink, useLocation } from 'react-router-dom';
 
-const useStyles = makeStyles(theme => ({
-    linkStyles: {
-        textDecoration: 'none',
-        color: 'black',
-        padding: '1em 0 0 2em',
-        fontSize: 18,
-        opacity: '50%',
-    },
-    profilePic: {
-        borderRadius: '50%',
-        width: '130px',
-        marginTop: '4em',
-    },
-    drawer: {
-        background: '#F9F9FC',
-    },
-    linkContainer: {
-        paddingTop: '2em',
-    },
-    profileName: {
-        paddingTop: '1em',
-        fontSize: 22,
-    },
-}));
+import profilePic from '../../static/images/profilePicSample.png';
+import { chatList, courseGroupList } from '../../data/mockData';
+import { useStyles } from './SidebarStyles';
+import AddIcon from '@material-ui/icons/Add';
+import RemoveIcon from '@material-ui/icons/Remove';
 
 const Drawer = props => {
     const classes = useStyles();
+    const location = useLocation();
+
+    const [courseList, setCourseList] = useState(
+        courseGroupList.map(cgl => ({ ...cgl, expand: false }))
+    );
+    const [chatId, setChatId] = useState(null);
+    const [courseId, setCourseId] = useState([]);
+
+    const showGroup = id => {
+        setCourseId([...courseId, id]);
+        const editedCourseList = courseList.map(cl => {
+            if (cl.id === id && !cl.expand) return { ...cl, expand: true };
+            if (cl.id === id && cl.expand) return { ...cl, expand: false };
+            else return { ...cl };
+        });
+        setCourseList(editedCourseList);
+    };
+
+    if (location.pathname === '/chat')
```

Comment:
I think instead of putting the contents of the sidebar in this Component, and checking react-router for the path, the Sidebar component could be a wrapper (which has the styling necessary to be mobile responsive and whatnot), and the sidebar content can be a separate component that is passed as a child of the Sidebar.  Thoughts?

Reply:
I think that would be cleaner and easy to handle the general stying of the sidebar. 
But even it is structured like that, I still need to use react-router to check for the path inside the child components? Or is there any other way to display the correct content to the right page?